File: modules/apps/change-tracking/change-tracking-rest-client-js/src/models/CTProcess.ts
/**
 * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

			import {Status} from './Status';

/**
 * @author David Truong
 * @generated
 */

	/**
	* Represents publications that have been published.
	*/
	export class CTProcess {
			"actions"?: {[key: string]: {[key: string]: string;};};
			"ctCollectionId"?: number;
			"datePublished"?: Date;
			"description"?: string;
			"id"?: number;
			"name"?: string;
			"ownerName"?: string;
			"status"?: Status;

		static "discriminator": string | undefined = undefined;

	static "attributeTypeMap": Array<{
		baseName: string;
		name: string;
		type: string;
	}> = [
		{
			baseName: "actions",
			name: "actions",
			type: "{[key: string]: {[key: string]: string;};}",
		},
		{
			baseName: "ctCollectionId",
			name: "ctCollectionId",
			type: "number",
		},
		{
			baseName: "datePublished",
			name: "datePublished",
			type: "Date",
		},
		{
			baseName: "description",
			name: "description",
			type: "string",
		},
		{
			baseName: "id",
			name: "id",
			type: "number",
		},
		{
			baseName: "name",
			name: "name",
			type: "string",
		},
		{
			baseName: "ownerName",
			name: "ownerName",
			type: "string",
		},
		{
			baseName: "status",
			name: "status",
			type: "Status",
		},
		];

		static getAttributeTypeMap() {
				return CTProcess.attributeTypeMap;
		}
	}

File: modules/apps/change-tracking/change-tracking-api/src/main/java/com/liferay/change/tracking/model/CTProcess.java
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.change.tracking.model;

import com.liferay.portal.kernel.annotation.ImplementationClassName;
import com.liferay.portal.kernel.model.PersistedModel;
import com.liferay.portal.kernel.util.Accessor;

import org.osgi.annotation.versioning.ProviderType;

/**
 * The extended model interface for the CTProcess service. Represents a row in the &quot;CTProcess&quot; database table, with each column mapped to a property of this class.
 *
 * @author Brian Wing Shun Chan
 * @see CTProcessModel
 * @generated
 */
@ImplementationClassName("com.liferay.change.tracking.model.impl.CTProcessImpl")
@ProviderType
public interface CTProcess extends CTProcessModel, PersistedModel {

	/*
	 * NOTE FOR DEVELOPERS:
	 *
	 * Never modify this interface directly. Add methods to <code>com.liferay.change.tracking.model.impl.CTProcessImpl</code> and rerun ServiceBuilder to automatically copy the method declarations to this interface.
	 */
	public static final Accessor<CTProcess, Long> CT_PROCESS_ID_ACCESSOR =
		new Accessor<CTProcess, Long>() {

			@Override
			public Long get(CTProcess ctProcess) {
				return ctProcess.getCtProcessId();
			}

			@Override
			public Class<Long> getAttributeClass() {
				return Long.class;
			}

			@Override
			public Class<CTProcess> getTypeClass() {
				return CTProcess.class;
			}

		};

}
// SB-Hash:2109382993